File: gerenciador-certificado-api/src/main/java/br/com/gerenciadorcertificadoapi/services/CertificadoPJService.java
package br.com.gerenciadorcertificadoapi.services;

import br.com.gerenciadorcertificadoapi.data.vo.CertificadoPJVO;
import br.com.gerenciadorcertificadoapi.models.CertificadoPJ;
import br.com.gerenciadorcertificadoapi.models.enums.TipoCertificado;

import java.util.List;

public interface CertificadoPJService {

    List<CertificadoPJVO> findAllPaginado(TipoCertificado tipoCertificado, int page, int itens);
    List<CertificadoPJVO> findAll(TipoCertificado tipoCertificado);
    List<CertificadoPJVO> findAllExpired(TipoCertificado tipoCertificado);
    CertificadoPJVO getById(String uuid);
    CertificadoPJVO create(CertificadoPJVO certificadoVO);
    void delete(String uuid);
    CertificadoPJVO findByCnpj(String cnpj, TipoCertificado tipo);
    List<CertificadoPJVO> findByRazaoSocial(String razaoSocial, TipoCertificado tipo);
    long countAll();

}

File: gerenciador-certificado-ui/src/resources/certificado-pj/certificado-pj.service.ts
import { CertificadoPJ, CertificadoPJResponse } from './certificado-pj.resources';
import { useAuth } from '@/resources';

class CertificadoPJService {
    baseUrl: string = process.env.NEXT_PUBLIC_API_URL + '/api/certificado-pj';
    auth = useAuth();

    async getAllByTipo(tipo: String, page: number, limit: number): Promise<CertificadoPJResponse> {
        const useSession = this.auth.getUserSession();
        const response = await fetch(`${this.baseUrl}/${tipo}?page=${page}&limit=${limit}`, {
            headers: {
                "Authorization": `Bearer ${useSession?.accessToken}`
            }
        });
        const data = await response.json();
        return {
            data: data.data.map((item: any) => ({
                uuid: item.uuid,
                razaoSocial: item.razaoSocial,
                cnpj: item.cnpj,
                dataEmissao: item.dataEmissao,
                dataVencimento: item.dataVencimento,
                tipoCertificado: item.tipoCertificado,
                valido: item.valido
            })),
            total: data.total
        };
    }

    async getByRazaoAndTipo(query: string, tipo: string): Promise<CertificadoPJ[]> {
        const useSession = this.auth.getUserSession();
        const response = await fetch(`${this.baseUrl}/find-razao-social?razaoSocial=${query}&tipoCertificado=${tipo}`, {
            headers: {
                "Authorization": `Bearer ${useSession?.accessToken}`
            }
        });
        return await response.json();
    }

    async getByCnpjAndTipo(cnpj: string, tipo: string): Promise<CertificadoPJ> {
        const useSession = this.auth.getUserSession();
        const response = await fetch(`${this.baseUrl}/find-cnpj?cnpj=${cnpj}&tipoCertificado=${tipo}`, {
            headers: {
                "Authorization": `Bearer ${useSession?.accessToken}`
            }
        });
        return await response.json();
    }

    async create(data: FormData): Promise<CertificadoPJ> {
        const useSession = this.auth.getUserSession();
        const response = await fetch(`${this.baseUrl}/upload`, {
            method: 'POST',
            body: data,
            headers: {
                "Authorization": `Bearer ${useSession?.accessToken}`
            }
        });

        if(response.status !== 201) {
            const responseError = await response.json();
            console.log(responseError);
            throw new Error(responseError.error);
        }
        return await response.json();
    }

    async delete(uuid: string): Promise<void> {
        try {
            const useSession = this.auth.getUserSession();
            const response = await fetch(`${this.baseUrl}/${uuid}`, { 
                method: 'DELETE',
                headers: {
                    "Authorization": `Bearer ${useSession?.accessToken}`
                }
            });
            if (!response.ok) {
                throw new Error('Erro ao excluir o certificado');
            }
        } catch (error) {
            console.error('Erro ao excluir o certificado:', error);
        }
    }
}

export const useCertificadoPJService = () => new CertificadoPJService();
